internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -221,13 +221,21 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 		}
 	}
 
-	msList, err := r.getMachineSetsForDeployment(ctx, d)
+	machineSets, err := r.getMachineSetsForDeployment(ctx, d)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
 	if d.Spec.Paused {
-		return ctrl.Result{}, r.sync(ctx, d, msList)
+		return ctrl.Result{}, r.sync(ctx, d, machineSets)
+	}
+
+	machines, err := r.getMachinesForMachineDeployment(ctx, d)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := r.syncManagedLabelsInPlace(ctx, machineSets, machines, d.Spec.Template.Spec.NodeLabels); err != nil {
```

Comment:
q: What about annotations? I think we should have ~almost similar propagation rules for labels and annotations, otherwise it could be confusing for users